Build failure: postgresql16JitPackages.postgis

### Steps To Reproduce

Steps to reproduce the behavior:
1. use aarch64-apple-darwin23.6.0
2. nixpkgs-unstable commit 039b72d0c738c934e2e36d7fc5520d1b425287a6
3. build `postgresql_16_jit.withPackages (ps: [ ps.postgis ])`

### Build log

```
last 24 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/b6i4m8ncipqc9ia4yyjvd0m7hygh43d8-postgis-3.4.2.tar.gz
       > source root is postgis-3.4.2
       > setting SOURCE_DATE_EPOCH to timestamp 1707412498 of file postgis-3.4.2/doc/topology_comments.sql
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
       > Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
       > Running phase: configurePhase
       > fixing libtool script ./build-aux/ltmain.sh
       > ./configure
       > patching script interpreter paths in ./configure
       > ./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/iafzjk7zbkqaszqfp6n006vvxjrpn4f6-bash-5.2p32/bin/sh"
       > configure flags: --prefix=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2 --bindir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/bin --sbindir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/sbin --includedir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/include --oldincludedir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/include --mandir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/share/man --infodir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/share/info --docdir=/nix/store/5cxwrnnc5njhx12a01qiyr5w5fhh7nww-postgis-3.4.2-doc/share/doc/postgis --libdir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/lib --libexecdir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/libexec --localedir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/share/locale --datadir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/share/postgresql --datarootdir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/share/postgresql --bindir=/nix/store/f0wl7s7md2lcbyh8hji5d9yhqhbgchhy-postgis-3.4.2/bin --docdir=/nix/store/5cxwrnnc5njhx12a01qiyr5w5fhh7nww-postgis-3.4.2-doc/share/doc/postgis --with-gdalconfig=/nix/store/00syb4zf2wf1h69r0a0qifs730pjzd6v-gdal-minimal-3.9.2/bin/gdal-config --with-jsondir=/nix/store/pcp2x53q5cz04cag2nn29fdfnjaxvwyr-json-c-0.17-dev --disable-extension-upgrades-install
       > checking for a BSD-compatible install... /nix/store/1l41471x6rlf7l544s1bxkqqgpprd28m-coreutils-9.5/bin/install -c
       > checking build system type... aarch64-apple-darwin23.6.0
       > checking host system type... aarch64-apple-darwin23.6.0
       > checking how to print strings... printf
       > checking for gcc... clang
       > checking whether the C compiler works... no
       > configure: error: in `/private/tmp/nix-build-postgis-3.4.2.drv-0/postgis-3.4.2':
       > configure: error: C compiler cannot create executables
       > See `config.log' for more details
       > /nix/store/riapfbcz0gbwpw2fv8s19aiisiv5zgk0-stdenv-darwin/setup: line 236: postgresqlStop: command not found
       For full logs, run 'nix log /nix/store/p9w1wprfzlbhcaarkk1jgqiwzj4fx61s-postgis-3.4.2.drv'.
```

### Additional context

GH Job: https://github.com/openstreetmap-ng/openstreetmap-ng/actions/runs/10889150978/job/30215154539

### Notify maintainers

@autra
@imincik
@l0b0
@nh2
@nialov
@sikmir
@willcohen
@MarcWeber
@wolfgangwalther 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

*I don't have an easy access to a darwin machine.*

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/servers/sql/postgresql/generic.nix
+++ b/pkgs/servers/sql/postgresql/generic.nix
@@ -47,7 +47,7 @@ let
     pname = "postgresql";
 
     stdenv' =
-      if jitSupport then
+      if jitSupport && !stdenv.cc.isClang then
         overrideCC llvmPackages.stdenv (llvmPackages.stdenv.cc.override {
           # LLVM bintools are not used by default, but are needed to make -flto work below.
           bintools = llvmPackages.bintools;
@@ -126,7 +126,6 @@ let
     # and allows splitting them cleanly.
     env.CFLAGS = "-fdata-sections -ffunction-sections"
       + (if stdenv'.cc.isClang then " -flto" else " -fmerge-constants -Wl,--gc-sections")
-      + lib.optionalString (stdenv'.isDarwin && jitSupport) " -fuse-ld=lld"
       # Makes cross-compiling work when xml2-config can't be executed on the host.
       # Fixed upstream in https://github.com/postgres/postgres/commit/0bc8cebdb889368abdf224aeac8bc197fe4c9ae6
       + lib.optionalString (olderThan "13") " -I${libxml2.dev}/include/libxml2";

